lib/holidays/date_calculator/weekend_modifier.rb

```diff
@@ -16,24 +16,47 @@ def to_monday_if_sunday(date)
         date
       end
 
+      # if Christmas falls on a Sunday, move it to the next Tuesday (Boxing Day will go on Monday)
+      # if Christmas falls on a Saturday, move it to the next Monday (Boxing Day will be Sunday and potentially Tuesday)
+      # used as a callback function, if xmas is not observed on the 25th
+      def xmas_to_weekday_if_weekend(year)
+        to_tuesday_if_sunday_or_monday_if_saturday(Date.civil(year, 12, 25))
+      end
+
       # Move Boxing Day if it falls on a weekend, leaving room for Christmas.
       # Used as a callback function.
       def to_weekday_if_boxing_weekend(date)
         if date.wday == 6 || date.wday == 0
           date += 2
-        elsif date.wday == 1
+        elsif date.wday == 1 # https://github.com/holidays/holidays/issues/27
           date += 1
         end
 
         date
       end
 
+      # if Christmas falls on a Saturday, move it to the next Monday (Boxing Day will be Sunday and potentially Tuesday)
+      # if Christmas falls on a Sunday, move it to the next Tuesday (Boxing Day will go on Monday)
+      #
+      # if Boxing Day falls on a Saturday, move it to the next Monday (Christmas will go on Friday)
+      # if Boxing Day falls on a Sunday, move it to the next Tuesday (Christmas will go on Saturday & Monday)
+      def to_tuesday_if_sunday_or_monday_if_saturday(date)
```

Comment:
Another quick note: I think leaving this here is good but I might slightly refactor it into my work after your work is merged. I think I did something kind of similar so we might have some duplication in the end. I'll make sure that I clear it up when I am resolving conflicts.